### ISSUE TITLE
Version 2, support for AES

### ISSUE THREAD

**Issue by andrivet** [andrivet/ADVobfuscator#35]
I am working on a new version of ADVobfuscator. A complete rewrite based on C++ 20. It will support obfuscation of string like the existing version but will permit to use a combination of different algorithms and keys so it will be more resistant. It will also avoid using undefined behaviour and will thus be more reliable (data always obfuscated at compile time).

It will also introduce AES encryption (still at compile time, decryption at runtime) for big chucks of data (for ex. certificates with private keys, ...)

Currently, the obfuscation part is ready but not the AES one which has some bugs I need to solve. Then I need to write the documentation, package the code, etc.

Don't ask when it will be released. I don't know. When I consider it to be ready.

**Comment by caizhe666** [comment]
It has been 3 years since last commit, if you encountered any error, please create a new dev branch and let us help you.

**Comment by andrivet** [comment]
I do not encounter any error and I will not create a dev branch. This is simply not a priority. It will be publish when it will be ready.

**Comment by caizhe666** [comment]
Then may I ask you what are you waiting?

**Comment by andrivet** [comment]
To have some time for this project.

**Comment by caizhe666** [comment]
Oh I'm looking forward to receiving any commits. Thanks for your great work.

**Comment by andrivet** [comment]
Why is this important for you? For what do you need this obfuscator?

**Comment by caizhe666** [comment]
I'm a Chinese and my project have to deal many crackers, by obfuscating I can protect it and prevent it being recognized as a malware (most Chinese  anti virus recognize vmp or other shells    as a malware)

**Comment by andrivet** [comment]
Interesting. Thank you for sharing.

**Comment by caizhe666** [comment]
Yes, there are many crackers in China, lol.

**Comment by Gmailaccount43y5fe** [comment]
> I am working on a new version of ADVobfuscator. A complete rewrite based on C++ 20. It will support obfuscation of string like the existing version but will permit to use a combination of different algorithms and keys so it will be more resistant. It will also avoid using undefined behaviour and will thus be more reliable (data always obfuscated at compile time).
> 
> It will also introduce AES encryption (still at compile time, decryption at runtime) for big chucks of data (for ex. certificates with private keys, ...)
> 
> Currently, the obfuscation part is ready but not the AES one which has some bugs I need to solve. Then I need to write the documentation, package the code, etc.
> 
> Don't ask when it will be released. I don't know. When I consider it to be ready.

**Comment by Gmailaccount43y5fe** [comment]
Hu

**Comment by Gmailaccount43y5fe** [comment]
Huuj

**Comment by Amadeus-** [comment]
Sadly, many of us can't progress to C++20 (or even close) due to APIs and other things we have to work with.  Many libraries aren't updated to even C++11 let alone C++20, and probably won't ever be.

**Comment by andrivet** [comment]
Yes, this is true (i am in such a case). So the library will not be usable. There is nothing I can do. I will not try to backport to earlier versions of the standard, it does not make sense.

**Comment by andrivet** [comment]
Some new about the progress: I have a working version 2.0 for Clang and GCC (Linux and macOS). I am now working on porting it to Visual Studio 22 (Windows) as this platform (Windows) is somewhat different than the others (int and long do not have the same length).
